src/fe_modes.c

```diff
@@ -105,29 +142,51 @@ int free_line_arrows_to_char(int last_arrow, int current_arrow) {
  *
  * Default mode. Used to choose other modes.
  */
-int mode_picker(State *state, WINDOW *canvas_win, WINDOW *status_win) {
-  // Mode Switch - Enter to canvas,
-  if (state->ch_in == KEY_ENTER) {
-    state->current_mode = state->last_canvas_mode;
+int mode_picker(reason_t reason, State *state) {
+  if (reason == END) {
     return 0;
   }
 
-  // LR Arrows navigation
+  int mode_start = MODE_PICKER + 1;
```

Comment:
```suggestion
// mode_start and num_modes are the bounds of the Mode_ID, used for enumerating all modes. 
  int mode_start = MODE_PICKER + 1;
```

Maybe call it `mode_first`